consensus/core/src/commit_finalizer.rs

```diff
@@ -608,15 +622,15 @@ impl CommitFinalizer {
                 continue;
             }
             let curr_block_state = blocks_map.get(&curr_block_ref).unwrap_or_else(|| panic!("Block {curr_block_ref} is either incorrectly gc'ed or failed to be recovered after crash.")).read();
-            // The first ancestor of current block should have the same origin / author as the current block.
-            // If it is not found in the blocks map but have round higher than the pending block, it might have
-            // voted on the pending block but have been GC'ed.
-            // Because the GC'ed block might have voted on the pending block and rejected some of the pending transactions,
-            // we cannot assume current block is voting to accept transactions from the pending block.
-            let curr_origin_ancestor_ref = curr_block_state.block.ancestors().first().unwrap();
-            let skip_votes = curr_block_ref.author == curr_origin_ancestor_ref.author
-                && pending_block_ref.round < curr_origin_ancestor_ref.round
-                && !blocks_map.contains_key(curr_origin_ancestor_ref);
+            // When proposing the current block, if it is possible that the pending block has already been GC'ed,
```

Comment:
nit: Instead of saying "proposing" block can we say "counting votes" or "counting tx votes", it seems clearer as to what we are doing here